xline/src/storage/kvstore.rs

```diff
@@ -35,8 +38,74 @@ pub(crate) struct KvStore {
 /// KV store inner
 #[derive(Debug)]
 struct KvStoreInner {
-    /// B-tree map
-    btree: Mutex<BTreeMap<Vec<u8>, KeyValue>>,
+    /// Key Index
+    index: Mutex<BTreeMap<Vec<u8>, Vec<KeyRevision>>>,
+    /// DB to store key value
+    db: Mutex<HashMap<Revision, KeyValue>>,
+    /// Revision
+    revision: Mutex<i64>,
+}
+
+/// Revison of a key
+#[derive(Debug, Copy, Clone)]
+struct KeyRevision {
+    /// Last creation revision
+    create_revision: i64,
+    /// Number of modification since last creation
+    version: i64,
+    /// Last modification revision
+    mod_revision: i64,
+    /// Sub revision in one transaction
+    sub_revision: i64,
```

Comment:
Do different sub revisions mean different key revision? If yes, a transaction may take many revisions. Is it what we're expecting?

Reply:
Operations in on txn will have the same global revision but different sub_revision. It doesn't mean different global revision.